src/Mapping/RequestParameterMapping.php

```diff
@@ -82,6 +82,13 @@ public function map(ServerRequestInterface $request, ResponseInterface $response
 					// Obtain request parameter values
 					$value = $requestParameters[$parameter->getName()];
 
+					if ($value === NULL && $parameter->isRequired()) {
+						throw new InvalidStateException(sprintf('Parameter "%s" should be provided in request attributes', $parameter->getName()));
+					}
+					if ($value === '' && !$parameter->isAllowEmpty()) {
+						throw new InvalidStateException(sprintf('Parameter "%s" should not be empty', $parameter->getName()));
+					}
+
 					// Normalize value
 					$normalizedValue = $mapper->normalize($value);
```

Comment:
Just small question.

What if we handle null value here and don't pass it to mapper. I'm not sure what's better. But these conditions `if ($vlaue === null) return null` in each mapper looks strange. Don't you think?

Reply:
I was thinking about the possibility to handle this in mapper. For example, if you have a complicated type (eg. like a composition of simple types).

Then there is a use case: create own complicated type (like DTO) with your representation of `null` value.

Reply:
I see. Seems reasonable.